R/rd-html.R

```diff
@@ -204,13 +204,21 @@ as_html.tag_Sexpr <- function(x, ...) {
   on.exit(setwd(old_wd), add = TRUE)
 
   # Environment shared across a file
-  res <- eval(parse(text = code), context_get("sexpr_env"))
+  outlines <- utils::capture.output({
+    out <- withVisible(eval(parse(text = code), context_get("sexpr_env")))
+    res <- out$value
+    if (out$visible)
+      print(res)
+  })
 
   results <- options$results %||% "rd"
   switch(results,
     text = as.character(res),
     rd = flatten_text(rd_text(as.character(res))),
     hide = "",
+    verbatim = paste0("<pre>\n",
+                      paste0(escape_html(outlines), collapse = "\n"),
+                      "\n</pre>\n"),
     cli::cli_abort(
```

Comment:
I think it'd be worth updating this error message since we now support all known forms of `\Sexpr`

Reply:
I have updated the error message as suggested. Note, however, that it is currently unreachable: other values not (partially) matching one of the 4 types would already fail from `parse_Rd()`.